modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -307,85 +322,114 @@ package object widgets {
     }
   }
 
-  abstract class Chart[C:ToPoints:Sampler] extends JsWorld[Seq[(String, Any)], Seq[(String, Any)]] {
-    import notebook.JSBus._
-
-    def originalData:C
-    def maxPoints:Int
-
+  abstract class OriginalDataToSeqConversions[C: ToPoints : Sampler](originalData: C, maxPoints: Int)
```

Comment:
Nice separation of concerns, but not sure about the name here. `OriginalDataToSeqConverter` ?